client/src/pages/LoginOrSignUp.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useContext } from "react";
+import { Link } from "react-router-dom";
+import { Container, Paper, Typography, makeStyles } from "@material-ui/core";
+import { AuthContext } from "../context/AuthContext";
+import FormInput from "../components/FormInput";
+import GenericButton from "../components/GenericButton";
+
+function LoginOrSignUp({ type }) {
+  const classes = useStyles();
+  const { errors, registerUser, loginUser } = useContext(AuthContext);
```

Comment:
I think you can create a custom hook to handle all your forms. For example 
```
import { useState } from "react";

export const useForm = initialValues => {
  const [values, setValues] = useState(initialValues);

  return [
    values,
    e => {
      setValues({
        ...values,
        [e.target.name]: e.target.value
      });
    }
  ];
};
```

And the use it like 
`const [values, setValues] = useForm({email:'', password:''})` for login page
`const [values, setValues] = useForm({name: '', email:'', password:''})` for signup page

Reply:
I like this idea, so I'll try implementing it later today.